server/channels/app/post.go

```diff
@@ -2144,9 +2144,30 @@ func (a *App) GetEditHistoryForPost(postID string) ([]*model.Post, *model.AppErr
 		}
 	}
 
+	if appErr := a.populateEditHistoryFileMetadata(posts); appErr != nil {
+		return nil, appErr
+	}
+
 	return posts, nil
 }
 
+func (a *App) populateEditHistoryFileMetadata(editHistoryPosts []*model.Post) *model.AppError {
+	for _, post := range editHistoryPosts {
+		fileInfos, err := a.Srv().Store().FileInfo().GetByIds(post.FileIds, true, true)
```

Comment:
Making a DB call in a loop seems like it could be a bit of a performance issue, but I suppose this shouldn't be called often enough to make a noticeable difference

Reply:
The cache saves some hit to the DB, but we need to get all the items - either in one big query or is multiple small queries. Bot are equivalent and considering the number of times we'd need to do this (very few), the simplest solution is the most effective.